components/raftstore/src/store/peer.rs

```diff
@@ -734,6 +746,7 @@ impl Peer {
                 self.pending_messages.push(resp);
                 return Ok(());
             }
+            self.should_wake_up = state == LeaseState::Expired;
```

Comment:
I think leader should wake up once it receives `MsgReadIndex`, not only when lease has expired.

Reply:
If it's suspect, it's woken up already. See L2281.